include/zephyr/net/wifi.h

```diff
@@ -25,24 +25,197 @@ enum wifi_security_type {
+	case WIFI_FREQ_BAND_5_GHZ:
+		return "5GHz";
+	case WIFI_FREQ_BAND_6_GHZ:
+		return "6GHz";
+	default:
+		return "UNKNOWN";
+	}
+}
+
 #define WIFI_SSID_MAX_LEN 32
 #define WIFI_PSK_MAX_LEN 64
 #define WIFI_MAC_ADDR_LEN 6
 
 #define WIFI_CHANNEL_MAX 233
 #define WIFI_CHANNEL_ANY 255
 
+/* Based on https://w1.fi/wpa_supplicant/devel/defs_8h.html#a4aeb27c1e4abd046df3064ea9756f0bc */
+enum wifi_iface_state {
+	WIFI_DISCONNECTED = 0,
+	WIFI_INTERFACE_DISABLED,
+	WIFI_INACTIVE,
+	WIFI_SCANNING,
+	WIFI_AUTHENTICATING,
+	WIFI_ASSOCIATING,
+	WIFI_ASSOCIATED,
+	WIFI_4WAY_HANDSHAKE,
+	WIFI_GROUP_HANDSHAKE,
+	WIFI_COMPLETED
+};
+
+/**
+ * wifi_state_txt - Get the connection state name as a text string
+ */
+static inline const char *wifi_state_txt(enum wifi_iface_state state)
+{
+	switch (state) {
+	case WIFI_DISCONNECTED:
+		return "DISCONNECTED";
+	case WIFI_INACTIVE:
+		return "INACTIVE";
+	case WIFI_INTERFACE_DISABLED:
+		return "INTERFACE_DISABLED";
+	case WIFI_SCANNING:
+		return "SCANNING";
+	case WIFI_AUTHENTICATING:
+		return "AUTHENTICATING";
+	case WIFI_ASSOCIATING:
+		return "ASSOCIATING";
+	case WIFI_ASSOCIATED:
+		return "ASSOCIATED";
+	case WIFI_4WAY_HANDSHAKE:
+		return "4WAY_HANDSHAKE";
+	case WIFI_GROUP_HANDSHAKE:
+		return "GROUP_HANDSHAKE";
+	case WIFI_COMPLETED:
+		return "COMPLETED";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+/* Based on https://w1.fi/wpa_supplicant/devel/structwpa__ssid.html#a625821e2acfc9014f3b3de6e6593ffb7 */
+enum wifi_iface_mode {
+	WIFI_MODE_INFRA = 0,
+	WIFI_MODE_IBSS = 1,
+	WIFI_MODE_AP = 2,
+	WIFI_MODE_P2P_GO = 3,
+	WIFI_MODE_P2P_GROUP_FORMATION = 4,
+	WIFI_MODE_MESH = 5,
+};
+
+/**
+ * wifi_mode_txt - Get the interface mode type as a text string
+ */
+static inline const char *wifi_mode_txt(enum wifi_iface_mode mode)
+{
+	switch (mode) {
+	case WIFI_MODE_INFRA:
+		return "STATION";
+	case WIFI_MODE_IBSS:
+		return "ADHOC";
+	case WIFI_MODE_AP:
+		return "ACCESS POINT";
+	case WIFI_MODE_P2P_GO:
+		return "P2P GROUP OWNER";
+	case WIFI_MODE_P2P_GROUP_FORMATION:
+		return "P2P GROUP FORMATION";
+	case WIFI_MODE_MESH:
+		return "MESH";
+	default:
+		return "UNKNOWN";
+	}
+}
+
+/* As per https://en.wikipedia.org/wiki/Wi-Fi#Versions_and_generations */
+enum wifi_link_mode {
+	WIFI_0 = 0,
+	WIFI_1,
+	WIFI_2,
+	WIFI_3,
+	WIFI_4,
+	WIFI_5,
+	WIFI_6,
+	WIFI_6E,
+	WIFI_7
+};
+
+/**
+ * wifi_link_mode_txt - Get the link mode type as a text string
+ */
+static inline const char *wifi_link_mode_txt(enum wifi_link_mode link_mode)
+{
+	switch (link_mode) {
+	case WIFI_0:
+		return "WIFI 0 (802.11)";
+	case WIFI_1:
+		return "WIFI 1 (802.11b)";
+	case WIFI_2:
+		return "WIFI 2 (802.11a)";
+	case WIFI_3:
+		return "WIFI 3 (802.11g)";
+	case WIFI_4:
+		return "WIFI 4 (802.11n/HT)";
+	case WIFI_5:
+		return "WIFI 5 (802.11ac/VHT)";
+	case WIFI_6:
+		return "WIFI 6 (802.11ax/HE)";
+	case WIFI_6E:
+		return "WIFI 6E (802.11ax 6GHz/HE)";
+	case WIFI_7:
+		return "WIFI 7 (802.11be/EHT)";
+	default:
+		return "UNKNOWN";
```

Comment:
Should we have a dedicated enum value for "UNKNOWN"? I suspect that some WiFi offloaded drivers won't have that information.

Reply:
We could, I am just using `-1` as the return value in case this is not filled.

Reply:
Even if we have dedicated enum for UNKNOWN, we still need to handle `default` case, so, are you suggesting to use `WIFI_LINK_MODE_UNKNOWN->"UNKNOWN"` and then `default->"not available"`?

Reply:
I think I understand now, I have removed `default` and added UNKNOWN and MAX ENUMs for all, please check.

Reply:
urgh..checkpatch is throwing a warning : `../../../../../../include/zephyr/net/wifi.h:157:1: error: control reaches end of non-void function [-Werror=return-type]`, so, need a `default` case, but what should be the return value `-1`, or just remove `case UNKNOWN`?

Reply:
What you have done with:
```
	case WIFI_SECURITY_TYPE_UNKNOWN:
	default:
		return "UNKNOWN";
```
look good to me.

My main point was about returning UNKNOWN enum instead of -1 by drivers implementing this interface. Thanks for changes!